qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -442,6 +442,14 @@ def __call__(self, target, basis_fidelity=None):
         return_circuit.compose(decomposition_euler[2*best_nbasis], [q[0]], inplace=True)
         return_circuit.compose(decomposition_euler[2*best_nbasis+1], [q[1]], inplace=True)
 
+        return_operator = Operator(return_circuit).data
+        for idx, elt in np.ndenumerate(return_operator):
+            if abs(elt) > 1e-8:
```

Comment:
Is there a reason you went for `1e-8` here? The normal default tolerance we use elsewhere are on the order of `1e-15`, `1e-12`, and `1e-13` for example:

https://github.com/Qiskit/qiskit-terra/blob/b9027930e014acc822798ff68baca9b5564c85bd/qiskit/quantum_info/synthesis/two_qubit_decompose.py#L95
https://github.com/Qiskit/qiskit-terra/blob/b9027930e014acc822798ff68baca9b5564c85bd/qiskit/quantum_info/synthesis/two_qubit_decompose.py#L41
https://github.com/Qiskit/qiskit-terra/blob/b9027930e014acc822798ff68baca9b5564c85bd/qiskit/quantum_info/synthesis/two_qubit_decompose.py#L69

Reply:
This is a good question. The approach and tolerances here are drawn from what we use for `Operator.equiv` ( https://github.com/Qiskit/qiskit-terra/blob/a0e7dcec145db9bd3ef7e5bf1566c23292db31a1/qiskit/quantum_info/operators/predicates.py#L64 ).

To @levbishop 's comment about preserving precision, I think we'd want the tolerance to be larger to decrease the likelihood we find an element that's order epsilon. We could scan the `np.abs` of the full array, and use the largest element from there, or maybe increase the tolerance to the smallest possible value that can be the largest magnitude in a unitary matrix, which I think is at least 1/n<sup>2</sup>, but maybe could be higher.